src/irmin/node.ml

```diff
@@ -143,9 +143,16 @@ struct
         let e = List.map to_entry e in
         Some (of_entries e)
 
-  let to_elt t = `Node (list t)
-  let of_values ~depth:_ l = Some (of_list l)
-  let of_inode ~depth:_ ~length:_ _ = None
+  type pre_elt =
+    [ `Node of (step * value) list | `Inode of int * (int * hash) list ]
+  [@@deriving irmin]
+
+  type elt =
+    [ `Node of (step * value) list | `Inode of int * (int list * hash) list ]
+  [@@deriving irmin]
```

Comment:
would it be possible to remove these two types to simplify the backendsdefinitions?

Reply:
Remove the `pre_elt` type here https://github.com/mirage/irmin/pull/1686/commits/59b8252e4e563b3054f2e80f18c21647606ba54d